DistroLauncher/RootUser.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "stdafx.h"
+
+RootSession::RootSession(HRESULT* hr_ptr)
+{
+    hr = hr_ptr;
+
+    ULONG distributionVersion;
+    PSTR* defaultEnvironmentVariables = nullptr;
+    ULONG defaultEnvironmentVariableCount = 0;
+
+    *hr = g_wslApi.WslGetDistributionConfiguration(&distributionVersion, &defaultUID, &wslDistributionFlags,
+                                                   &defaultEnvironmentVariables, &defaultEnvironmentVariableCount);
+
+    if (FAILED(*hr) || distributionVersion == 0) {
+        Helpers::PrintErrorMessage(*hr);
+        return;
+    }
+
+    // discarding the env variable string array otherwise they would leak.
+    for (ULONG i = 0; i < defaultEnvironmentVariableCount; ++i) {
+        CoTaskMemFree(static_cast<LPVOID>(defaultEnvironmentVariables[i]));
+    }
+
+    // Switching to ROOT
+    const ULONG uid = DistributionInfo::QueryUid(L"root");
+    *hr = g_wslApi.WslConfigureDistribution(uid, wslDistributionFlags);
```

Comment:
root uid is always 0, so no need to query it.

Reply:
Make sense. Fixed.